pyramid/request.py

```diff
@@ -214,10 +213,8 @@ def response(self):
         right" attributes (e.g. by calling ``request.response.set_cookie()``)
         within a view that uses a renderer.  Mutations to this response object
         will be preserved in the response sent to the client."""
-        registry = self.registry
-        response_factory = registry.queryUtility(IResponseFactory,
-                                                 default=Response)
-        return response_factory()
+        response_factory = _get_response_factory(self.registry)
+        return response_factory(self)
```

Comment:
Custom responses classes were not officially supported in 1.5. Pyramid documents all public APIs in the API section of the documentation.

Note the difference from http://docs.pylonsproject.org/projects/pyramid/en/1.5-branch/api/interfaces.html to http://docs.pylonsproject.org/projects/pyramid/en/1.6-branch/api/interfaces.html with regards to `IResponseFactory`.